lnwallet/musig_session_test.go

```diff
@@ -0,0 +1,247 @@
+	// and the sig in the same message.
+	h.refreshSession(nodeName, nextNonce, true)
+
+	return nextVerificationNonce
+}
+
+// ProcessVerificationNonce processes a verification nonce from the remote.
+// This is equivalent to receiving the revoke from a remote party after you
+// kicked off the commitment dance.
+func (h *muSessionHarness) ProcessVerificationNonce(nodeName nodeType,
+	nextNonce *musig2.Nonces) {
+
+	h.refreshSession(nodeName, nextNonce, true)
+}
+
+func newMuSessionHarness(t *testing.T) *muSessionHarness {
+	aliceCommit := wire.NewMsgTx(2)
+	aliceCommit.AddTxIn(&wire.TxIn{})
+
+	bobCommit := wire.NewMsgTx(2)
+	bobCommit.AddTxIn(&wire.TxIn{})
+
+	alicePriv, alicePub := btcec.PrivKeyFromBytes(testWalletPrivKey)
+	aliceSigner := input.NewMockSigner([]*btcec.PrivateKey{alicePriv}, nil)
+
+	aliceVerificationNonce, err := musig2.GenNonces(
+		musig2.WithPublicKey(alicePub),
+	)
+	require.NoError(t, err)
+
+	bobPriv, bobPub := btcec.PrivKeyFromBytes(bobsPrivKey)
+	bobSigner := input.NewMockSigner([]*btcec.PrivateKey{bobPriv}, nil)
+
+	bobVerificationNonce, err := musig2.GenNonces(
+		musig2.WithPublicKey(bobPub),
+	)
+	require.NoError(t, err)
+
+	inputTxOut := &wire.TxOut{
+		Value:    1000,
+		PkScript: testHdSeed[:],
+	}
+
+	aliceSession := NewMusigPairSession(&MusigSessionCfg{
+		LocalKey: keychain.KeyDescriptor{
+			PubKey: alicePub,
+		},
+		RemoteKey: keychain.KeyDescriptor{
+			PubKey: bobPub,
+		},
+		LocalNonce:  *aliceVerificationNonce,
+		RemoteNonce: *bobVerificationNonce,
+		Signer:      aliceSigner,
+		InputTxOut:  inputTxOut,
+	})
+
+	bobSession := NewMusigPairSession(&MusigSessionCfg{
+		LocalKey: keychain.KeyDescriptor{
+			PubKey: bobPub,
+		},
+		RemoteKey: keychain.KeyDescriptor{
+			PubKey: alicePub,
+		},
+		LocalNonce:  *bobVerificationNonce,
+		RemoteNonce: *aliceVerificationNonce,
+		Signer:      bobSigner,
+		InputTxOut:  inputTxOut,
+	})
+
+	return &muSessionHarness{
+		aliceCommit:  aliceCommit,
+		aliceSession: aliceSession,
+		bobCommit:    bobCommit,
+		bobSession:   bobSession,
+		t:            t,
+	}
+}
+
+// TestMusigSession tests that we're able to send and receive signatures for
+// the set of asymmetric musig sessions. This tests proper nonce rotation and
+// signature verification.
+func TestMusigSesssion(t *testing.T) {
+	t.Parallel()
+
+	// First, we'll make a new musig session between Alice and Bob. This is
+	// 4 sessions total, as both sides maintain a session for their local
+	// commitment, and one for the remote commitment.
+	muSessions := newMuSessionHarness(t)
+
+	const numRounds = 10
+	for i := 0; i < numRounds; i++ {
+		// We'll now simulate a full commitment dance.
+		//
+		// To start, Alice will sign a new commitment for Bob's remote
+		// commitment.
+		aliceSig := muSessions.SignCommitment(nodeAlice)
+
+		// Bob will then verify Alice's signature, and sign a new
+		// commitment for Alice.
+		bobSig, bobNonce := muSessions.VerifyAndSignCommitment(
```

Comment:
perhaps s/bobNonce/bobNextNonce? just to show that it is not the nonce for the returned sig

Reply:
Had that, but that goes over the line limit ;)